File: Python/ByRound/2102/2102_B_The_Picky_Cat.py
# Problem: CF 2102 B - The Picky Cat
# https://codeforces.com/contest/2102/problem/B

# B. The Picky Cat
# Purpose: Determine if the first element of an array can become the median after flipping signs of elements.
# Algorithm: 
# 1. For each test case, sort the absolute values of the array.
# 2. Check how many elements have absolute value less than the first element's absolute value.
# 3. If more than half of the elements are smaller in absolute value, then flipping signs may allow first element to become median.
# Time Complexity: O(n log n) due to sorting
# Space Complexity: O(n) for storing input array

for s in [*open(0)][2::2]:  # Read input lines starting from index 2, stepping by 2 (even indices)
    a = (*map(abs, map(int, s.split())),)  # Convert string line to tuple of absolute values
    print("YNEOS"[sum(x < a[0] for x in a) * 2 > len(a) :: 2])  # Evaluate condition and print YES/NO


# https://github.com/VaHiX/codeForces/

File: CSharp/ByRound/2102/2102_B_The_Picky_Cat.cs
// Problem: CF 2102 B - The Picky Cat
// https://codeforces.com/contest/2102/problem/B

/*
B. The Picky Cat

Algorithm/Technique: Sorting, Greedy, Median Calculation

Time Complexity: O(n log n) per test case due to sorting (implicit in median calculation)
Space Complexity: O(n) for storing the array elements

The problem asks if we can make the first element of the array become the median
by flipping signs of any elements. The key idea is:
1. We compare absolute values of all elements except the first one.
2. Count how many of those have absolute value less than that of the first element.
3. If this count is at most half of the array size (rounded down), then we can flip signs
   to make the first element the median.

*/
using System;
using System.Collections.Generic;
class Program {
  static void Main() {
    int testCases = int.Parse(Console.ReadLine());
    while (testCases-- > 0) {
      int arraySize = int.Parse(Console.ReadLine());
      int[] numbers = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
      int referenceValue = Math.Abs(numbers[0]); // Absolute value of first element
      int countLessThanReference = 0; // Count of elements whose absolute value is less than reference
      
      // Loop through all elements except the first one
      for (int index = 1; index < arraySize; index++) {
        if (Math.Abs(numbers[index]) < referenceValue)
          countLessThanReference++;
      }
      
      // If count of smaller elements is <= half the array size, we can make first element median
      if (countLessThanReference <= arraySize / 2)
        Console.WriteLine("YES");
      else
        Console.WriteLine("NO");
    }
  }
}


// https://github.com/VaHiX/codeForces/